homeassistant/helpers/template.py

```diff
@@ -2112,6 +2113,65 @@ def as_timedelta(value: str) -> timedelta | None:
     return dt_util.parse_duration(value)
 
 
+def merge_response(
+    value: ServiceResponse, sort_by: str | None = None, selected_key: str | None = None
+) -> list[Any]:
+    """Merge action responses into single list.
+
+    Checks that the input is a correct service response:
+    {
+        "entity_id": {str: dict[str, Any]},
+    }
+    If response is a list, it will extend the list with the items.
+    """
+    if not isinstance(value, dict):
+        raise TypeError("Response is not a dictionary")
+    if not value:
+        # Bail out early if response is empty empty
+        return []
+
+    is_list = False
+    response_items: list = []
+    for entity_id, entity_response in value.items():
+        if not isinstance(entity_response, dict):
+            raise TypeError("Response is not a dictionary")
+        for value_key, type_response in entity_response.items():
+            if len(entity_response) == 1 and isinstance(type_response, list):
+                is_list = True
+                for dict_in_list in type_response:
+                    if isinstance(dict_in_list, dict):
+                        dict_in_list["entity_id"] = entity_id
+                        dict_in_list["value_key"] = value_key
+                response_items.extend(type_response)
+            else:
+                break  # Don't continue looping if not a list
```

Comment:
Please explain why we break out of the `for value_key, type_response in entity_response.items()` loop

Reply:
Provided explanation (as logic is handled below for when it's not a single list with dicts.